lib/pbench/test/unit/server/query_apis/test_index_mappings.py

```diff
@@ -0,0 +1,190 @@
+                            "controller": {"type": "keyword"},
+                            "date": {"type": "date"},
+                            "end": {"type": "date"},
+                            "id": {"type": "keyword"},
+                            "iterations": {"type": "text"},
+                            "name": {"type": "keyword"},
+                            "script": {"type": "keyword"},
+                            "start": {"type": "date"},
+                            "toolsgroup": {"type": "keyword"},
+                            "user": {"type": "keyword"},
+                        }
+                    },
+                    "sosreports": {
+                        "type": "nested",
+                        "properties": {
+                            "hostname-f": {"type": "keyword"},
+                            "hostname-s": {"type": "keyword"},
+                            "inet": {
+                                "type": "nested",
+                                "properties": {
+                                    "ifname": {"type": "keyword"},
+                                    "ipaddr": {"type": "ip"},
+                                },
+                            },
+                            "inet6": {
+                                "type": "nested",
+                                "properties": {
+                                    "ifname": {"type": "keyword"},
+                                    "ipaddr": {"type": "keyword"},
+                                },
+                            },
+                            "md5": {"type": "keyword"},
+                            "name": {"type": "keyword"},
+                            "sosreport-error": {"type": "text"},
+                        },
+                    },
+                },
+            },
+            version=5,
+        )
+
+    with monkeypatch.context() as m:
+        m.setattr(Template, "find", fake_find)
+        yield
+
+
+@pytest.fixture()
+def database_error(monkeypatch):
+    def raise_db_error(name: str):
+        raise DatabaseError("DB Error")
+
+    with monkeypatch.context() as m:
+        m.setattr(Template, "find", raise_db_error)
+        yield
+
+
+class TestIndexMappings:
+    """
+    Unit testing for resources/IndexMappings class.
+
+    In a web service context, we access class functions mostly via the
+    Flask test client rather than trying to directly invoke the class
+    constructor and `post` service.
+    """
+
+    def test_query(self, client, server_config, run_template):
+        """
+        Check the construction of index mappings API and filtering of the
+        response body.
+        """
+        with client:
+            response = client.get(f"{server_config.rest_uri}/index/mappings")
+            assert response.status_code == HTTPStatus.OK
+            res_json = response.json
+            assert isinstance(res_json, dict)
+            assert len(res_json.keys()) == 4
```

Comment:
I'd like to verify the specific keys we expect, and the structure, rather than just the number of top-level keys.